core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -307,7 +307,7 @@ private[spark] class KryoSerializerInstance(ks: KryoSerializer) extends Serializ
   override def deserialize[T: ClassTag](bytes: ByteBuffer): T = {
     val kryo = borrowKryo()
     try {
-      input.setBuffer(bytes.array)
+      input.setBuffer(bytes.array(), bytes.arrayOffset() + bytes.position(), bytes.remaining())
```

Comment:
I'm saying that the change I proposed is not necessary, not that your change is not necessary.